beacon_node/network/src/beacon_processor/mod.rs

```diff
@@ -554,11 +737,151 @@ impl<T: BeaconChainTypes> GossipProcessor<T> {
+
+                            let block = Box::new(verified_block.block.clone());
+                            match chain.process_block(verified_block) {
+                                Ok(_block_root) => {
+                                    metrics::inc_counter(
+                                        &metrics::BEACON_PROCESSOR_GOSSIP_BLOCK_IMPORTED_TOTAL,
+                                    );
+
+                                    trace!(
+                                        log,
+                                        "Gossipsub block processed";
+                                        "peer_id" => peer_id.to_string()
+                                    );
+
+                                    // TODO: It would be better if we can run this _after_ we publish the block to
+                                    // reduce block propagation latency.
+                                    //
+                                    // The `MessageHandler` would be the place to put this, however it doesn't seem
+                                    // to have a reference to the `BeaconChain`. I will leave this for future
+                                    // works.
+                                    match chain.fork_choice() {
+                                        Ok(()) => trace!(
+                                            log,
+                                            "Fork choice success";
+                                            "location" => "block gossip"
+                                        ),
+                                        Err(e) => error!(
+                                            log,
+                                            "Fork choice failed";
+                                            "error" => format!("{:?}", e),
+                                            "location" => "block gossip"
+                                        ),
+                                    }
+                                }
+                                Err(BlockError::ParentUnknown { .. }) => {
+                                    // Inform the sync manager to find parents for this block
+                                    // This should not occur. It should be checked by `should_forward_block`
+                                    error!(
+                                        log,
+                                        "Block with unknown parent attempted to be processed";
+                                        "peer_id" => peer_id.to_string()
+                                    );
+                                    send_sync_message(
+                                        sync_tx,
+                                        SyncMessage::UnknownBlock(peer_id, block),
+                                        &log,
+                                    );
+                                }
+                                other => {
+                                    warn!(
```

Comment:
Can we suppress `BlockAlreadyKnown` here as well? We get a lot of confused users asking about it on Discord and it's really spammy (it could be a debug log IMO)